app/src/main/java/com/sschuraytz/charcoaldrawing/DrawingView.java

```diff
@@ -42,13 +45,27 @@ public boolean onTouchEvent(MotionEvent event) {
 
         switch(event.getAction()) {
             case MotionEvent.ACTION_DOWN:
-                printCircleWithLocation(pointX, pointY);
+                if (isEraseMode)
```

Comment:
If we add another Tool (like SmudgeTool) how does this code have to change?

Reply:
I could make an enum of tools and then have a current tool that is set to one of the enum options.

Reply:
> If we add another Tool (like SmudgeTool) how does this code have to change?

I could make an enum of tools and then have a current tool that is set to one of the enum options.

Reply:
You don't need an enum of tools, you have a bunch of Tool classes. You just need to keep track of whatever the current Tool is which is a reference to the Tool that the user wants to use.

Reply:
> You don't need an enum of tools, you have a bunch of Tool classes. You just need to keep track of whatever the current Tool is which is a reference to the Tool that the user wants to use.

I added a currentTool variable. I wanted to make a method that accepts a tool and sets currentTool to that. But, in the classes where I use the method, I don't have access to the tools. So, instead, I made a separate method for setting each tool. (line 132-138 in DrawingView, used on lines 85 & 94 in Main Activity) Does that make sense?